sbom-utility-scripts/scripts/base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -87,23 +114,77 @@ def get_base_images_sbom_components(base_images_digests, is_last_from_scratch):
     return components
 
 
+def get_base_images_from_dockerfile(parsed_dockerfile):
+    """
+    Reads the base images from provided parsed dockerfile
+
+    :param parsed_dockerfile: (Dict) - Contents of the parsed dockerfile
+    :return: base_images (List) - List of base images used during build as extracted
+                                  from the dockerfile in the order they were used.
+
+    Example:
+    If the Dockerfile looks like
+    FROM registry.access.redhat.com/ubi8/ubi:latest as builder
+    ...
+    FROM builder
+    ...
+
+    Then the relevant part of parsed_dockerfile look like
+    {
+        "Stages": [
+            {
+                "BaseName": "registry.access.redhat.com/ubi8/ubi:latest",
+                "As": "builder",
+                "From": {"Image": "registry.access.redhat.com/ubi8/ubi:latest"},
+            },
+            {
+                "BaseName": "builder",
+                "From": {"Stage": {"Named": "builder", "Index": 0}},
+            },
+        ]
+    },
+    """
+    base_images = []
+
+    # this part of the json is the relevant one that contains the
+    # info about base images
+    stages = parsed_dockerfile["Stages"]
+
+    for stage in stages:
+        if "Image" in stage["From"]:
+            base_images.append(stage["From"]["Image"])
+        elif "Scratch" in stage["From"]:
+            base_images.append("scratch")
+        elif "Stage" in stage["From"]:
+            stage_index = stage["From"]["Stage"]["Index"]
+            # Find the original stage/image. Named stage can refer to another named stage,
+            # so continue looking until we find the image those stages refer to.
+            while stage_index is not None and stage_index != {}:
+                refered_stage = stages[stage_index]
+                stage_index = refered_stage.get("From").get("Stage", {}).get("Index", None)
+                if stage_index is None or stage_index == {}:
```

Comment:
nitpick: I don't think `stage_index` will ever be a dict
```suggestion
            while stage_index is not None:
                refered_stage = stages[stage_index]
                stage_index = refered_stage.get("From").get("Stage", {}).get("Index", None)
                if stage_index is None:
```

Reply:
you are right, the following `.get` handles that and it will be just None. Good catch, updated.